cmd/heimdallcli/stake.go

```diff
@@ -48,11 +55,26 @@ func StakeCmd() *cobra.Command {
 				return err
 			}
 
+			route := fmt.Sprintf("custom/%s", checkpointEndpoint)
+
+			bz, _, err := cliCtx.QueryWithData(route, nil)
+			if err != nil {
+				return err
+			}
+
+			var params chainmanagerTypes.Params
+			json.Unmarshal(bz, &params)
+
+			stakingManagerAddress := params.ChainParams.StakingManagerAddress.EthAddress()
+			stakeManagerInstance, err := contractCaller.GetStakeManagerInstance(stakingManagerAddress)
+
 			return contractCaller.StakeFor(
```

Comment:
This api is changed. we need to pass Pubkey as well.

Reply:
`StakeFor` internally fetch pubkey